lib/Core/Inference/FrameBuilder/AssignmentWalker.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace Phpactor\WorseReflection\Core\Inference\FrameBuilder;
+
+use Phpactor\WorseReflection\Core\Inference\FrameWalker;
+use Microsoft\PhpParser\Node;
+use Phpactor\WorseReflection\Core\Inference\Frame;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Microsoft\PhpParser\Node\Expression\ListIntrinsicExpression;
+use Microsoft\PhpParser\Node\Expression\MemberAccessExpression;
+use Microsoft\PhpParser\Node\Expression\SubscriptExpression;
+use Phpactor\WorseReflection\Core\Inference\SymbolContext;
+use Phpactor\WorseReflection\Core\Inference\FrameBuilder;
+use Microsoft\PhpParser\Node\Expression\AssignmentExpression;
+use Phpactor\WorseReflection\Core\Logger;
+use Phpactor\WorseReflection\Core\Inference\SymbolFactory;
+use Phpactor\WorseReflection\Core\Inference\Symbol;
+use Phpactor\WorseReflection\Core\Inference\Variable as WorseVariable;
+use Microsoft\PhpParser\Token;
+use Phpactor\WorseReflection\Core\Type;
+use Microsoft\PhpParser\Node\ArrayElement;
+
+class AssignmentWalker implements FrameWalker
```

Comment:
This class should probably issue command back to the FrameBuilder or delegate to it's own "walkers".